src/sidebar/util/state-util.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+/**
+ * Return a value from app state when it meets certain criteria.
+ *
+ * `await` returns a Promise which resolves when a selector function,
+ * which reads values from a Redux store, returns non-null.
+ *
+ * @param {Object} store - Redux store
+ * @param {Function<T|null>} selector - Function which returns a value from the
+ *   store if the criteria is met or `null` otherwise.
+ * @return {Promise<T>}
+ */
+function awaitStateChange(store, selector) {
+  var result = selector(store);
+  if (result !== null) {
+    return Promise.resolve(result);
+  }
+  return new Promise(resolve => {
+    var unsubscribe = store.subscribe(() => {
+      var result = selector(store);
+      if (result !== null) {
+        unsubscribe();
+        resolve(result);
+      }
+    });
+  });
+}
+
+module.exports = {
+  awaitStateChange: awaitStateChange,
```

Comment:
In object literals in new code, I've been using [object literal shorthand](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand) to shorten `{ foo: foo }` to `{ foo }`. This isn't mandatory, but I find it a little more readable.